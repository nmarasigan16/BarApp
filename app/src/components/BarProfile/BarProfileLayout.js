import React, { Component } from 'react';
import {
    View,
    FlatList,
} from 'react-native';
import CoverComponent from './CoverComponent';
import RatioComponent from './RatioComponent';
import SpecialRowComponent from "./SpecialRowComponent";
import Text from '../general/text/TextComponent';
import {colorScheme} from "../../lib/styles/ColorScheme";



const BarProfileLayout = ({bar}) => {
    // TODO link up with api
    const specials = [
        {
            id: 1,
            description: '$3 drafts.  Includes Legend\'s Lager, Guinness, Blue Moon, Coors Light, Miller Lite, and more!',
            object: {
                name: 'Rail Card'
            }
        },
        {
            id: 2,
            description: '$2 Tequila shots'
        }
    ];
    return (
        <View>
            <View
                style={{
                    flexDirection: 'row',
                    alignItems: 'center',
                    justifyContent: 'center',
                }}>
                <CoverComponent cover={10}/>
                <RatioComponent guys={50}/>
            </View>
            <View
                style={{
                    flexDirection: 'column',
                    marginTop: 20,
                    margin: 10
                }}>
                <Text
                    style={{
                        alignSelf: 'center',
                        textAlign: 'center',
                        color: colorScheme.accent,
                        fontWeight: 'bold'
                    }}>
                    Specials
                </Text>
                <FlatList
                    data={specials}
                    keyExtractor={(item) => {
                        return item.id;
                    }}
                    renderItem={({item}) => <SpecialRowComponent special={item}/>}
                />
            </View>
        </View>
    )
};

export default BarProfileLayout;