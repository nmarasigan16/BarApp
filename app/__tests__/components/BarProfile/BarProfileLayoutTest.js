import 'react-native';
import React from 'react';
import BarProfileLayout from '../../../src/components/BarProfile/BarProfileLayout';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <BarProfileLayout/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
