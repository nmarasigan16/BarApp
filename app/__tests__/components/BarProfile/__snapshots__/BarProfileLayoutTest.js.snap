// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "silver",
                "fontSize": 30.607554623001164,
                "margin": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Cover
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "silver",
                "fontSize": 45.911331934501746,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          $10
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Courier New",
              },
              Object {
                "color": "red",
                "marginTop": 5,
              },
            ]
          }
        >
          report
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "silver",
              "fontSize": 30.607554623001164,
              "marginBottom": 10,
            },
          ]
        }
      >
        Ratio
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "Courier New",
                },
                Object {
                  "color": "deepskyblue",
                  "fontSize": 45.911331934501746,
                },
              ]
            }
          >
            50
            %
          </Text>
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "deepskyblue",
                  "fontSize": 30.607554623001164,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "Courier New",
                },
                Object {
                  "color": "deeppink",
                  "fontSize": 45.911331934501746,
                },
              ]
            }
          >
            50
            %
          </Text>
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "deeppink",
                  "fontSize": 30.607554623001164,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 10,
        "marginTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Courier New",
          },
          Object {
            "alignSelf": "center",
            "color": "silver",
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Specials
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "description": "$3 drafts.  Includes Legend's Lager, Guinness, Blue Moon, Coors Light, Miller Lite, and more!",
            "id": 1,
            "object": Object {
              "name": "Rail Card",
            },
          },
          Object {
            "description": "$2 Tequila shots",
            "id": 2,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "margin": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Courier New",
                      },
                      Object {
                        "color": "silver",
                      },
                    ]
                  }
                >
                  $3 drafts.  Includes Legend's Lager, Guinness, Blue Moon, Coors Light, Miller Lite, and more!
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Courier New",
                      },
                      Object {
                        "color": "darkgray",
                        "fontStyle": "italic",
                      },
                    ]
                  }
                >
                  View 
                  Rail Card
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "margin": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginLeft": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Courier New",
                      },
                      Object {
                        "color": "silver",
                      },
                    ]
                  }
                >
                  $2 Tequila shots
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
